src/avro/src/schema.rs

```diff
@@ -2328,6 +2344,94 @@ mod tests {
                 namespace: "name.has".into()
             }
         );
+
+        // Use default namespace when namespace is not provided.
+        // Materialize uses "" as the default namespace.
+        let schema = Schema::parse_str(
+            r#"{"type": "record", "name": "TestDoc", "doc": "Doc string",
+            "fields": [{"name": "name", "type": "string"}]}"#,
+        )
+        .unwrap();
+        assert_eq!(schema.named.len(), 1);
+        assert_eq!(
+            schema.named[0].name,
+            FullName {
+                name: "TestDoc".into(),
+                namespace: "".into()
+            }
+        );
+
+        // Empty namespace strings should be allowed.
+        let schema = Schema::parse_str(
+            r#"{"type": "record", "namespace": "", "name": "TestDoc", "doc": "Doc string",
+            "fields": [{"name": "name", "type": "string"}]}"#,
+        )
+        .unwrap();
+        assert_eq!(schema.named.len(), 1);
+        assert_eq!(
+            schema.named[0].name,
+            FullName {
+                name: "TestDoc".into(),
+                namespace: "".into()
+            }
+        );
+
+        // Equality of names is defined on the FullName and is case-sensitive.
+        let first = Schema::parse_str(
+            r#"{"type": "fixed", "namespace": "namespace",
+            "name": "name", "size": 1}"#,
+        )
+        .unwrap();
+        let second = Schema::parse_str(
+            r#"{"type": "fixed", "name": "namespace.name",
+            "size": 1}"#,
+        )
+        .unwrap();
+        assert_eq!(first.named[0].name, second.named[0].name);
+
+        let first = Schema::parse_str(
+            r#"{"type": "fixed", "namespace": "namespace",
+            "name": "name", "size": 1}"#,
+        )
+        .unwrap();
+        let second = Schema::parse_str(
+            r#"{"type": "fixed", "name": "namespace.Name",
+            "size": 1}"#,
+        )
+        .unwrap();
+        assert_ne!(first.named[0].name, second.named[0].name);
+
+        let first = Schema::parse_str(
+            r#"{"type": "fixed", "namespace": "Namespace",
+            "name": "name", "size": 1}"#,
+        )
+        .unwrap();
+        let second = Schema::parse_str(
+            r#"{"type": "fixed", "namespace": "namespace",
+            "name": "name", "size": 1}"#,
+        )
+        .unwrap();
+        assert_ne!(first.named[0].name, second.named[0].name);
+
+        // The name portion of a fullname, record field names, and enum symbols must:
+        // start with [A-Za-z_] and subsequently contain only [A-Za-z0-9_]
+        assert!(Schema::parse_str(
+            r#"{"type": "record", "name": "99 problems but a name aint one",
```

Comment:
Lol, did you invent this name or did it come from upstream

Reply:
Since I went to the trouble of looking this up: https://github.com/apache/avro/pull/718 :D

Reply:
It came from upstream and I didn't want the joke to die! https://github.com/apache/avro/blob/master/lang/py/avro/test/test_schema.py#L394